In [2]:
# load data
library(haven)
wage1 <- read_dta("data_ps1/wage1.dta")

In [3]:
colnames(wage1)

[1] "wage"     "educ"     "exper"    "tenure"   "nonwhite" "female"  
 [7] "married"  "numdep"   "smsa"     "northcen" "south"    "west"    
[13] "construc" "ndurman"  "trcommpu" "trade"    "services" "profserv"
[19] "profocc"  "clerocc"  "servocc"  "lwage"    "expersq"  "tenursq"

In [4]:
summary(wage1$educ)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   0.00   12.00   12.00   12.56   14.00   18.00 

1a. the avg education level in the sample is just shy of 13 years. the lowest is 0 and highest 18.

In [5]:
summary(wage1$wage)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.530   3.330   4.650   5.896   6.880  24.980 

1b. the avg hourly wage is 5.896 USD, which seems a bit low.

In [6]:
CPI_1976 <- 56.9
CPI_2003 <- 184.0

delta_CPI <- CPI_2003 / CPI_1976
delta_CPI

wage1$wage_2003 = (184 / 56.9) * wage1$wage
summary(wage1$wage_2003)

[1] 3.233743

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  1.714  10.768  15.037  19.066  22.248  80.779 

yes, it sems more reasonable, the mean in 19 USD.

In [7]:
t <- table(wage1$female)

paste("Male: ", t[1])
paste("Female: ", t[2])

# proportion
paste("Male: ", t[1]/nrow(wage1))
paste("Female: ", t[2]/nrow(wage1))

[1] "Male:  274"

[1] "Female:  252"

[1] "Male:  0.520912547528517"

[1] "Female:  0.479087452471483"

In [8]:
meap01 <- read_dta("data_ps1/meap01.dta")
colnames(meap01)
summary(meap01)


[1] "dcode"   "bcode"   "math4"   "read4"   "lunch"   "enroll"  "expend" 
 [8] "exppp"   "lenroll" "lexpend" "lexppp"

     dcode           bcode          math4            read4       
 Min.   : 1010   Min.   :   1   Min.   :  0.00   Min.   :  0.00  
 1st Qu.:33902   1st Qu.:1386   1st Qu.: 61.60   1st Qu.: 48.90  
 Median :55100   Median :2794   Median : 76.40   Median : 62.70  
 Mean   :53028   Mean   :3395   Mean   : 71.91   Mean   : 60.06  
 3rd Qu.:77010   3rd Qu.:5084   3rd Qu.: 87.00   3rd Qu.: 73.90  
 Max.   :83070   Max.   :8838   Max.   :100.00   Max.   :100.00  
     lunch            enroll           expend            exppp      
 Min.   :  0.00   Min.   :  62.0   Min.   : 275985   Min.   : 1207  
 1st Qu.: 16.61   1st Qu.: 287.0   1st Qu.:1474465   1st Qu.: 4502  
 Median : 35.10   Median : 379.0   Median :1894043   Median : 5078  
 Mean   : 39.25   Mean   : 401.9   Mean   :2036984   Mean   : 5195  
 3rd Qu.: 59.06   3rd Qu.: 484.0   3rd Qu.:2407636   3rd Qu.: 5767  
 Max.   :100.00   Max.   :1496.0   Max.   :7665998   Max.   :11958  
    lenroll         lexpend          lexppp     
 Min. 

the largest val is 100, the smallest is 0. Sure.

In [9]:
numb_max <- sum(meap01$math4==100)

numb_max/length(meap01$dcode) # percentage of total schools

[1] 0.02084476

In [10]:
sum(meap01$math4==50)

[1] 17

In [11]:
mean(meap01$math4)
mean(meap01$read4)

[1] 71.909

[1] 60.06188

In [12]:
cor(meap01$math4, meap01$read4)

[1] 0.8427281

I can conclude that those with high pass percentages in math also tend to have high pass percentages in reading. The correlation is 0.84, which is quite high.

In [23]:
m <- mean(meap01$exppp)
std <- sd(meap01$exppp)

round(std/m*100, 2)

[1] 21.02

yes there is a wide variation in per pupil spending, around 21% of the mean.

# Part 2 

In [25]:
ceosal2 <- read_dta("data_ps1/CEOSAL2.dta")

In [27]:
colnames(ceosal2)

[1] "salary"   "age"      "college"  "grad"     "comten"   "ceoten"  
 [7] "sales"    "profits"  "mktval"   "lsalary"  "lsales"   "lmktval" 
[13] "comtensq" "ceotensq" "profmarg"

In [26]:
summary(ceosal2)

     salary            age           college            grad       
 Min.   : 100.0   Min.   :33.00   Min.   :0.0000   Min.   :0.0000  
 1st Qu.: 471.0   1st Qu.:52.00   1st Qu.:1.0000   1st Qu.:0.0000  
 Median : 707.0   Median :57.00   Median :1.0000   Median :1.0000  
 Mean   : 865.9   Mean   :56.43   Mean   :0.9718   Mean   :0.5311  
 3rd Qu.:1119.0   3rd Qu.:62.00   3rd Qu.:1.0000   3rd Qu.:1.0000  
 Max.   :5299.0   Max.   :86.00   Max.   :1.0000   Max.   :1.0000  
     comten         ceoten           sales          profits      
 Min.   : 2.0   Min.   : 0.000   Min.   :   29   Min.   :-463.0  
 1st Qu.:12.0   1st Qu.: 3.000   1st Qu.:  561   1st Qu.:  34.0  
 Median :23.0   Median : 6.000   Median : 1400   Median :  63.0  
 Mean   :22.5   Mean   : 7.955   Mean   : 3529   Mean   : 207.8  
 3rd Qu.:33.0   3rd Qu.:11.000   3rd Qu.: 3500   3rd Qu.: 208.0  
 Max.   :58.0   Max.   :37.000   Max.   :51300   Max.   :2700.0  
     mktval         lsalary          lsales          lmktval  

In [33]:
lsalary <- log(ceosal2$salary)
lsales <- log(ceosal2$sales)
lmktval <- log(ceosal2$mktval)

In [35]:
m <- lm(lsalary ~ lsales + lmktval, data=ceosal2)
summary(m)


Call:
lm(formula = lsalary ~ lsales + lmktval, data = ceosal2)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.28060 -0.31137 -0.01269  0.30645  1.91210 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)  4.62092    0.25441  18.163  < 2e-16 ***
lsales       0.16213    0.03967   4.087 6.67e-05 ***
lmktval      0.10671    0.05012   2.129   0.0347 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5103 on 174 degrees of freedom
Multiple R-squared:  0.2991,	Adjusted R-squared:  0.2911 
F-statistic: 37.13 on 2 and 174 DF,  p-value: 3.727e-14


The constant elasticity equation is 
$$
\log(salary) = 4.62 + .162\log(sales) + .107\log(mktval)
$$
$n=177$, $R^2=0.299$

The variables (sales and market value) explain about 29.9% of the variation in CEO salaries. The rest (70.1%) is due to other factors not in the model, randomness, measurement error, etc.

An increase of 1% in sales is associated with an increase of 0.162% in salary, ceteris paribus.

In [36]:
m2 <- lm(lsalary ~ lsales + lmktval + profits, data=ceosal2)
summary(m2)


Call:
lm(formula = lsalary ~ lsales + lmktval + profits, data = ceosal2)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.27002 -0.31026 -0.01027  0.31043  1.91489 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 4.687e+00  3.797e-01  12.343  < 2e-16 ***
lsales      1.614e-01  3.991e-02   4.043 7.92e-05 ***
lmktval     9.753e-02  6.369e-02   1.531    0.128    
profits     3.566e-05  1.520e-04   0.235    0.815    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5117 on 173 degrees of freedom
Multiple R-squared:  0.2993,	Adjusted R-squared:  0.2872 
F-statistic: 24.64 on 3 and 173 DF,  p-value: 2.53e-13


We cannot include profits in logarithmic form because profits are negative for 
nine of the companies in the sample. There is a very slight increase to the $R^2$, thus it is safe to assume that the addition of profits does not improve the model significantly.

In [39]:
m3 <- lm(lsalary ~ lsales + lmktval + profits + ceoten, data=ceosal2)
summary(m3)


Call:
lm(formula = lsalary ~ lsales + lmktval + profits + ceoten, data = ceosal2)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.48792 -0.29369  0.00827  0.29951  1.85524 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 4.558e+00  3.803e-01  11.986  < 2e-16 ***
lsales      1.622e-01  3.948e-02   4.109 6.14e-05 ***
lmktval     1.018e-01  6.303e-02   1.614   0.1083    
profits     2.905e-05  1.503e-04   0.193   0.8470    
ceoten      1.168e-02  5.342e-03   2.187   0.0301 *  
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5062 on 172 degrees of freedom
Multiple R-squared:  0.3183,	Adjusted R-squared:  0.3024 
F-statistic: 20.08 on 4 and 172 DF,  p-value: 1.387e-13


One more year as CEO increases predicted salary by 1.2%.

In [41]:
cor(ceosal2$lmktval,ceosal2$profits)

[1] 0.7768976

The sample correlation between log(mktval) and profits is about .78, which is 
fairly high.  As we know, this causes no bias in the OLS estimators, although it can 
cause their variances to be large.  Given the fairly substantial correlation between 
market value and firm profits, it is not too surprising that the latter adds nothing to 
explaining CEO salaries.  Also, profits is a short term measure of how the firm is 
doing while mktval is based on past, current, and expected future profitability. 

In [42]:
bwght2 <- read_dta("data_ps1/BWGHT2.dta")

In [44]:
lbwght <- log(bwght2$bwght)

In [47]:
m <- lm(lbwght ~ cigs + npvis, data = bwght2)
summary(m)


Call:
lm(formula = lbwght ~ cigs + npvis, data = bwght2)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.19743 -0.08369  0.02328  0.11336  0.44269 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  8.055451   0.016060 501.595  < 2e-16 ***
cigs        -0.003155   0.001161  -2.717  0.00666 ** 
npvis        0.005617   0.001310   4.289  1.9e-05 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.1994 on 1653 degrees of freedom
  (176 observations deleted due to missingness)
Multiple R-squared:  0.01591,	Adjusted R-squared:  0.01472 
F-statistic: 13.36 on 2 and 1653 DF,  p-value: 1.754e-06


In [55]:
sd_npvis<-sd(bwght2$npvis,na.rm = T)
b_npvis<-summary(m)$coefficients[3,1]

sd_npvis
b_npvis

sd_npvis*b_npvis*100 # percent increase in birthweight if there is one standard deviation increase in number of parental visits 

[1] 3.680791

[1] 0.005616643

[1] 2.067369

In [57]:
m2 <- lm(lbwght ~ cigs, data = bwght2)
summary(m2)


Call:
lm(formula = lbwght ~ cigs, data = bwght2)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.23489 -0.08507  0.01932  0.11840  0.43619 

Coefficients:
             Estimate Std. Error  t value Pr(>|t|)    
(Intercept)  8.120997   0.004940 1643.877  < 2e-16 ***
cigs        -0.003436   0.001133   -3.032  0.00246 ** 
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.1985 on 1720 degrees of freedom
  (110 observations deleted due to missingness)
Multiple R-squared:  0.005317,	Adjusted R-squared:  0.004739 
F-statistic: 9.194 on 1 and 1720 DF,  p-value: 0.002465


In [59]:
cor(bwght2$cigs,bwght2$npvis, use = "na.or.complete")

[1] -0.03872346

they are not very correlated, so beta 1 stays almost the same for simple and multiple regression.